### DISCUSSION TITLE
The suggestions in the quick switch menu are often bad, why?

### DISCUSSION THREAD

**Discussion by timjrobinson** [regolith-linux/regolith-desktop#834] [General]
I use the quick switch menu (special + space) for opening most apps, and often it gives really bad results causing me to open the wrong program. 

Some examples:

- When I type `sig` to open signal, the top search result is "Password and Keys", signal is second. 
- When I type `tele` to open Telegram, the top search result is "Skype", Telegram is second.

Why is this happening? Those first suggestions don't have the search in their name at all.

**Comment by fhchl** [comment]
I have the same issue. For example, sometimes, typing `code` gives Firefox as the top result.

**Comment by kgilmer** [comment]
When filtering from all desktop entries, ilia matches against the ["name" and "keywords" section of the desktop metadata](https://github.com/regolith-linux/ilia/blob/main/src/apps/DesktopAppPage.vala#L248).  My guess is that in these cases the matches are coming from the keywords section.  If available would you prefer a setting that limited filter results to name only?

```
$ cat /usr/share/applications/org.gnome.seahorse.Application.desktop | grep -i sig
Keywords=keyring;encryption;security;sign;ssh;gpg;pgp;
```

**Reply by fhchl** [reply]
Either such an option or a preference for when the entry matches the name would be awesome!

**Reply by timjrobinson** [reply]
Yea doing name only or at least name as a higher preference than keywords would be good. Could do all name matches followed by all keyword matches.

**Comment by solarkennedy** [comment]
This sounds like the same thing I was facing (typing in something exactly, and that exact thing was not the top sort), I fixed this bug in https://github.com/regolith-linux/ilia/pull/113
